src/syntax/rules.rkt

```diff
@@ -119,76 +111,40 @@
     (register-ruleset! 'name 'groups `((var . ,(get-representation 'type)) ...)
                        '((rname input output) ...))]))
 
-;; Templated rulesets defined by types. These are used to generate duplicate rules that
-;; are valid in any representation of the same underlying type.
-
-(define *templated-rulesets* (make-parameter '()))
-(define *templated-reprs* (make-parameter '()))
+(define (type-of-rule input output ctx)
```

Comment:
Similar question as about `repr-of-rule`. Uhh, to be honest I'm not sure why all this type-and-repr machinery still needs to exist in `rules.rkt`.

Reply:
Will check, but I believe each rule stores its output type/representation. It's definitely needed somewhere.

Reply:
Right, I understand that the output type is stored. What I'm not sure of is why `type-of-rule` embeds its own type checking logic; I feel like `type-of-rule` could just return `(type-of output ctx)`.

Reply:
Same as above.